src/modules/tools/zprobe/ZProbe.cpp

```diff
@@ -507,3 +516,63 @@ void ZProbe::home()
     Gcode gc(THEKERNEL->is_grbl_mode() ? "G28.2" : "G28", &(StreamOutput::NullStream));
     THEKERNEL->call_event(ON_GCODE_RECEIVED, &gc);
 }
+
+// measure repeatability of probe (M48)
+void ZProbe::repeatability(StreamOutput *stream, int number_point, float x, float y) {
+  bool probe_result;
+  float mm;
+  float sum = 0.0, mean = 0.0, sigma = 0.0, min = 99999.9, max = -99999.9, sample[number_point];
```

Comment:
All float constants MUST be appended with F, otherwise they incur a double to float conversion.
So 0.0F wherever they are used.